datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -293,14 +313,15 @@ func (f *ChainDataFetcher) resLoop() {
 			logger.Info("stopped resLoop for chaindatafetcher")
 			return
 		case res := <-f.resCh:
-			if res.err != nil {
-				logger.Error("db insertion is failed", "blockNumber", res.blockNumber, "reqType", res.reqType, "err", res.err)
-				// TODO-ChainDataFetcher add retry logic when data insertion is failed
-			} else {
-				if err := f.updateCheckpoint(res.blockNumber.Int64()); err != nil {
-					logger.Error("Failed to update checkpoint", "err", err, "checkpoint", res.blockNumber.Int64())
+			for {
```

Comment:
This loop is necessary? I think this is not necessary.

Reply:
Right. As you said, we can ignore saving checkpoint for a while. I removed the retry logic.